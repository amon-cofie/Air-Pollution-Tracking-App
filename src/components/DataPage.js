import React from 'react';
import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { useSelector } from 'react-redux';
import { fetchPollutionData } from '../redux/pollutionData';

const DataPage = ({ cityCoordinate }) => {
  const dispatch = useDispatch();
  const pollutionData = useSelector(
    (state) => state.pollutionData.pollutionData,
  );
  console.log(cityCoordinate);

  useEffect(() => {
    dispatch(fetchPollutionData(cityCoordinate));

    // fetchPollutionData(cityCoordinate);
  }, []);

  console.log(pollutionData);

  return (
    <div>
      <h1>Under Construction</h1>
    </div>
  );
};

export default DataPage;
