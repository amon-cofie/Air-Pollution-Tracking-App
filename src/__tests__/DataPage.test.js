import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import DataPage from '../components/DataPage';

test('City component renders correctly', () => {
  const cityCoordinate = {};
  const fetchedData = {
    pollution: {
      list: [{ components: {}, main: { aqi: 1 } }],
    },
  };
  const snap = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <DataPage cityCoordinate={cityCoordinate} pollutionData={fetchedData} />
      </BrowserRouter>
    </Provider>,
  );
  expect(snap).toMatchSnapshot();
});
