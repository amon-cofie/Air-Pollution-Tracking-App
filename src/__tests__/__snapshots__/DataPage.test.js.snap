// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City component renders correctly 1`] = `
<div
  className="mainDataContainer"
>
  <div
    className="prelimData"
  >
    <div
      className="prelimCityData"
    >
      <div
        className="cityAndCountry"
      >
        <p
          className="cityName"
        />
        <p
          className="cityCountry"
        >
           
          <span />
        </p>
      </div>
      <p
        className="cityState"
      />
    </div>
    <div
      className="cityAqiData"
    >
      AQI:
       
      1
    </div>
  </div>
  <div
    className="dataTableContainer"
  >
    <table
      className="dataTable"
    >
      <thead>
        <tr>
          <th>
            Pollutant
          </th>
          <th>
            Concentration
          </th>
          <th>
            Score
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            CO
          </td>
          <td />
          <td />
        </tr>
        <tr>
          <td>
            NH
            <sub>
              3
            </sub>
          </td>
          <td />
          <td />
        </tr>
        <tr>
          <td>
            NO
          </td>
          <td />
          <td>
            Rare pollutant
          </td>
        </tr>
        <tr>
          <td>
            NO
            <sub>
              2
            </sub>
          </td>
          <td />
          <td />
        </tr>
        <tr>
          <td>
            O
            <sub>
              3
            </sub>
          </td>
          <td />
          <td />
        </tr>
        <tr>
          <td>
            PM
            <sub>
              2.5
            </sub>
          </td>
          <td />
          <td />
        </tr>
        <tr>
          <td>
            PM
            <sub>
              10
            </sub>
          </td>
          <td />
          <td />
        </tr>
        <tr>
          <td>
            SO
            <sub>
              2
            </sub>
          </td>
          <td />
          <td />
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="coverFoot"
  />
</div>
`;
